workload/src/client.cc

```diff
@@ -604,12 +619,17 @@ x509svid_SVID *workloadapi_Client_FetchX509SVID(workloadapi_Client *client,
         if(*err != NO_ERROR) {
             return NULL;
         }
+        if(arrlen(svids) == 0) {
+            return NULL; // Should never happen
+        }
+        x509svid_SVID *ret_svid = svids[0];
+        arrfree(svids);  // free outer array
+        return ret_svid; // no response -> no bundle
+    } else {
+        // could not fetch x509 svid;
+        *err = ERROR1;
+        return NULL;
```

Comment:
We need to check if when we have a list with more than 1 svids, will cause a memory leak.